instrumentation/opentelemetry-instrumentation-pika/src/opentelemetry/instrumentation/pika/utils.py

```diff
@@ -0,0 +1,160 @@
+from opentelemetry.propagators.textmap import CarrierT, Getter
+from opentelemetry.semconv.trace import (
+    MessagingOperationValues,
+    SpanAttributes,
+)
+from opentelemetry.trace import Tracer
+from opentelemetry.trace.span import Span
+
+
+class _PikaGetter(Getter):  # type: ignore
+    def get(self, carrier: CarrierT, key: str) -> Optional[List[str]]:
+        value = carrier.get(key, None)
+        if value is None:
+            return None
+        return [value]
+
+    def keys(self, carrier: CarrierT) -> List[str]:
+        return []
+
+
+_pika_getter = _PikaGetter()
+
+
+def _decorate_callback(
+    callback: Callable[[Channel, Basic.Deliver, BasicProperties, bytes], Any],
+    tracer: Tracer,
+    task_name: str,
+):
+    def decorated_callback(
+        channel: Channel,
+        method: Basic.Deliver,
+        properties: BasicProperties,
+        body: bytes,
+    ) -> Any:
+        if not properties:
+            properties = BasicProperties()
+        span = _get_span(
+            tracer,
+            channel,
+            properties,
+            task_name=task_name,
+            operation=MessagingOperationValues.RECEIVE,
+        )
+        with trace.use_span(span, end_on_exit=True):
+            propagate.inject(properties.headers)
+            retval = callback(channel, method, properties, body)
+        return retval
+
+    return decorated_callback
+
+
+def _decorate_basic_publish(
+    original_function: Callable[[str, str, bytes, BasicProperties, bool], Any],
+    channel: Channel,
+    tracer: Tracer,
+):
+    def decorated_function(
+        exchange: str,
+        routing_key: str,
+        body: bytes,
+        properties: BasicProperties = None,
+        mandatory: bool = False,
+    ) -> Any:
+        if not properties:
+            properties = BasicProperties()
+        span = _get_span(
+            tracer,
+            channel,
+            properties,
+            task_name="(temporary)",
+            operation=None,
+        )
+        if not span:
+            return original_function(
+                exchange, routing_key, body, properties, mandatory
+            )
+        with trace.use_span(span, end_on_exit=True):
+            if span.is_recording():
+                propagate.inject(properties.headers)
+            retval = original_function(
+                exchange, routing_key, body, properties, mandatory
+            )
+        return retval
+
+    return decorated_function
+
+
+def _get_span(
+    tracer: Tracer,
+    channel: Channel,
+    properties: BasicProperties,
+    task_name: str,
+    operation: Optional[MessagingOperationValues] = None,
+) -> Optional[Span]:
+    if properties.headers is None:
+        properties.headers = {}
+    ctx = propagate.extract(properties.headers, getter=_pika_getter)
+    if context.get_value("suppress_instrumentation") or context.get_value(
+        _SUPPRESS_INSTRUMENTATION_KEY
+    ):
+        print("Suppressing instrumentation!")
```

Comment:
I don't believe any of our other instrumentations prints upon suppression.

Reply:
Removed :)